### ISSUE TITLE
Create graphs from experimental results (Iris data set and diabetes data set)

### ISSUE THREAD

**Issue by davidjwu** [chengsoonong/mclass-sky#126]
Compare the accuracy of the active learning algorithm using the (i) analytic center; (ii) Chebyshev center; and (iii) random center. Do this using the Iris data set and diabetes data set. 

For each data set produce a graph showing the performance of the algorithm.

**Comment by davidjwu** [comment]
@chengsoonong I've successfully generated a graph for the Iris data set. 

I've been having issues generating a graph for the diabetes data set, though. The diabetes data set is more complex (8 features as opposed to 2 features) and is more than twice as large. I've tried running the computation overnight but I don't think my computer is fast enough to do it in a timely manner. (It appears to be the rejection sampling causing the issue.) 

Ideally I'd like to collect data on the average accuracy taken over 25 tests for up to 100 iterations. For the moment, I am computing this data for 10 tests and 30 iterations, simply to have some data. (I suspect this should take a few hours). Perhaps I will try only using 4 of the features. Let me know if you have any thoughts.

**Comment by chengsoonong** [comment]
1. use all features.
2. collect statistics about the number of rejects (remember the count).
3. 10 tests with 30 iterations sound sensible.

### How to check that 30 iterations is sensible?
Compare it to the performance of logistic regression on a **randomly selected set** of examples, and check the test error on a hold out test set. How many examples to use in training?
1. 30 - the number you also have after 30 iterations
2. 50% of the training data (sensible because it leaves 50% in the test set)
3. 80% of the training data (same size as choosing 5 fold CV).

Repeat this test 10 times of course.

**Comment by davidjwu** [comment]
Thank you for the recommendations. 

I'll incorporate these suggestions into the notebook I am working on.

**Comment by davidjwu** [comment]
I ran the computation (10 tests with 30 iterations for each of the AC, CC and random center) for about 6 hours today and it didn't finish. 

I added some testing variables and ran the computation for the AC case and it appears millions of random vectors are being computed without the computation finishing, as we expected! (In the case of the Iris flower data set, the computation for any of the AC, CC and random center certainly required less than 1 million random vectors.)

**Comment by davidjwu** [comment]
For the moment I ran the computation with 10 tests and 15 iterations; which you can see in my GitHub repository. I'll run the computation overnight with 10 tests and 30 iterations and see how it goes.

**Comment by chengsoonong** [comment]
May I suggest running one test at a time? I mean 1 test for 30 iterations. See how long that takes, which then tells you how long 10 tests will take.

Since the tests are independent, you can start a new one by setting a different random seed.

**Comment by chengsoonong** [comment]
Graphs in commit 6ae94fb68fc35347982dd012eb9cfffe1e9eea0f resolve this issue.
